File: solutions/0874-backspace-string-compare/solution.py
class Solution(object):
    def backspaceCompare(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        ls1,ls2=[],[]
        for i in s:
            if i=='#' and len(ls1)>0:
                ls1.pop()
            elif i!='#':
                ls1.append(i)
        for i in t:
            if i=='#' and len(ls2)>0:
                ls2.pop()
            elif i!='#':
                ls2.append(i)
        print(ls1)
        print(ls2)
        return True if ls1==ls2 else False

File: solutions/0874-backspace-string-compare/solution.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        StringBuilder sc=new StringBuilder();
        for(char a:s.toCharArray())
        {
            if(a!='#')
                sc.append(a);
            else if(sc.length()>=1)
                sc.deleteCharAt(sc.length() - 1);
        }
        StringBuilder sc1=new StringBuilder();
        for(char a:t.toCharArray())
        {
            if(a!='#')
                sc1.append(a);
            else if(sc1.length()>=1)
                sc1.deleteCharAt(sc1.length() - 1);
        }
        return sc1.toString().equals(sc.toString());
    }
}
